### PR TITLE
fix(denols): remove unneeded lsp attach

### PR DESCRIPTION
**Pull request by sigmaSd**
fix https://github.com/neovim/nvim-lspconfig/issues/1998

I don't have a good idea *how* does this fixes the issue, but so far in my testing with this pr it seems to work perfectly.

### CODE REVIEW

File: lua/lspconfig/server_configurations/denols.lua [hunk 1]
```diff
@@ -35,7 +35,6 @@ local function virtual_text_document_handler(uri, result)
         vim.api.nvim_buf_set_option(bufnr, 'readonly', true)
         vim.api.nvim_buf_set_option(bufnr, 'modified', false)
         vim.api.nvim_buf_set_option(bufnr, 'modifiable', false)
-        lsp.buf_attach_client(bufnr, client_id)
```
Comment (justinmk) [inline lua/lspconfig/server_configurations/denols.lua:38]: linter noticed that `client_id` is not used now.

Comment (sigmaSd) [reply lua/lspconfig/server_configurations/denols.lua:38]: fixed

**Comment by sigmaSd** [comment]
The commit that added this https://github.com/neovim/nvim-lspconfig/pull/753

@kitagry can you confirm that this is not needed anymore?

**Comment by kitagry** [comment]
I added this code in order to use lsp functions (e.x. goto definition) in virtual text file. And, is it work correctly?

I'll see it this night. Thank you.

**Comment by sigmaSd** [comment]
Yes it seem to work inside virtual document without it.

**Comment by justinmk** [comment]
lint CI is failing because of wrong indentation

**Comment by kitagry** [comment]
Hmm, I can't reproduce this bug. Can you try with neovim develpment build?

NVIM v0.8.0-dev+1470-g0124a7bfa
Build type: Release
LuaJIT 2.1.0-beta3
nvim-lspconfig c55e830

> Yes it seem to work inside virtual document without it.

I think it works only when you specify `single_file_support=true`.

**Comment by sigmaSd** [comment]
> I think it works only when you specify single_file_support=true.

I can confirm this, do you have a suggestion on how to proceed then? is there a disadvantage of making `single_file_support=true` by default?

**Comment by kitagry** [comment]
> is there a disadvantage of making single_file_support=true by default?

I think there are no disadvantage. I think you should this change with `single_file_support=true` by default.

**Pull request merged**